File: SQL/Expressions/SQLLengthFunctionExpression.vb

' ___________________________________________________
'
'  © Hi-Integrity Systems 2012. All rights reserved.
'  www.hisystems.com.au - Toby Wicks
' ___________________________________________________
'

Option Strict On
Option Explicit On

Namespace SQL

    ''' <summary>
    ''' SQL function that returns the length of a string.
    ''' </summary>
    Public Class SQLLengthFunctionExpression
        Inherits SQLFunctionExpression

        Public Sub New(objExpression As SQLExpression)

            MyBase.New(objExpression)

        End Sub

        Protected Overrides Function FunctionName(ByVal eConnectionType As Database.ConnectionType) As String

            Select Case eConnectionType
                Case Database.ConnectionType.MySQL
                    Return "LENGTH"
                Case Database.ConnectionType.SQLServer, Database.ConnectionType.MicrosoftAccess
                    Return "LEN"
                Case Else
                    Throw New NotImplementedException(eConnectionType.ToString)
            End Select

            Return MyBase.FunctionName(eConnectionType)

        End Function

    End Class

End Namespace

File: SQL/Expressions/SQLLengthFunctionExpression.cs
// ___________________________________________________
//
//  © Hi-Integrity Systems 2012. All rights reserved.
//  www.hisystems.com.au - Toby Wicks
// ___________________________________________________
//

using System.Collections;
using System;
using System.Data;

namespace DatabaseObjects.SQL
{
	/// <summary>
	/// SQL function that returns the length of a string.
	/// </summary>
	public class SQLLengthFunctionExpression : SQLFunctionExpression
	{
		public SQLLengthFunctionExpression(SQLExpression objExpression) 
            : base(objExpression)
		{
		}
			
		internal override string SQL(Serializers.Serializer serializer)
		{
			return serializer.SerializeLengthFunctionExpression(this);
		}
	}
}
